src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -224,12 +226,40 @@ ss::future<> reconciler::commit_object(const object_range_info& range) {
      * TODO: we aren't actually replicating an overlay batch here yet. This will
      * come at the time of integration with the data layout STM.
      */
-    range.partition->lro = range.info.last_offset + model::offset(1);
+    const auto& part = range.partition->partition;
+    object_id id = object_id{uuid_t::create()};
+    dl_overlay_object overlay_obj(
+      id,
+      first_byte_offset_t{range.physical_offset_start},
+      byte_range_size_t{
+        range.physical_offset_end - range.physical_offset_start},
+      dl_stm_object_ownership::shared);
+
+    dl_overlay overlay{
+      range.info.base_offset,
+      range.info.last_offset,
+      range.info.base_timestamp,
+      range.info.last_timestamp,
+      range.info.terms,
+      overlay_obj};
+
+    auto push_result = co_await part->dl_stm_api()->push_overlay(overlay);
+    if (push_result.has_error()) {
+        vlog(
+          lg.error,
+          "Failed to replicate dl_overlay batch to {}, error: {}",
+          part->ntp(),
+          push_result.error());
+        co_return;
+    }
+
+    range.partition->lro = range.info.last_offset + kafka::offset(1);
```

Comment:
LRO is last reconciled offset so why it is `+1`? The `+1` belongs to `make_reader`. Or if we want it to start at `0` then it should be called `next_offset_to_reconcile`?